khal: update to 0.10.5.

dataCobra opened the pull request:
#### Testing the changes
- I tested the changes in this PR: **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64)
- I built this PR locally for these architectures (if supported. mark crossbuilds):
  - aarch64-musl (crossbuild)

Since version 0.10.5 there is no longer a `__khal` file. This is why I deleted line 26.

dataCobra commented:
Sadly the last PR #32292 doesn't explain why the make_check=ci_skip is added.

What do I need to do for xlint to not fail?

paper42 commented:
> Failed: DID NOT RAISE <class 'khal.khalendar.exceptions.CouldNotCreateDbDir'>

It seems like it's trying to chmod and create some directories in a place where it shouldn't be possible which succeeds in CI because there, it runs as root.

classabbyamp commented:
> CHANGE we are not shipping a zsh completion file anymore but provide documentation on how to generate completion files for bash, zsh, and fish (see the install section of the documentation)
>
> Packagers: please generate and ship those completion files if possible

`__khal` was removed because it got replaced by this mechanism

also, please add the changelog: https://raw.githubusercontent.com/pimutils/khal/master/CHANGELOG.rst

cc maintainer @Anachron

dataCobra commented:
> > CHANGE we are not shipping a zsh completion file anymore but provide documentation on how to generate completion files for bash, zsh, and fish (see the install section of the documentation)
> > Packagers: please generate and ship those completion files if possible
> 
> `__khal` was removed because it got replaced by this mechanism

Oh, that's what I was looking for yesterday. Thank you for finding it! I read two changelogs
yesterday about 0.10.5 and none of these had this info. Because of this I just removed the
file.

I'll try to get this implemented.

> 
> also, please add the changelog: https://raw.githubusercontent.com/pimutils/khal/master/CHANGELOG.rst

Where do you want me to add the CHANGELOG.rst? -> `usr/share/doc/`?

I've also tried to create the manpage, but I couldn't really figure out how to do it the best way.

classabbyamp commented:
just `changelog="url"` in the template

classabbyamp commented:
https://github.com/pimutils/khal/blob/master/doc/source/install.rst#shell-completion

just add the commands that redirect to a file, then vcomplete those files in post_install

dataCobra commented:
> https://github.com/pimutils/khal/blob/master/doc/source/install.rst#shell-completion
> 
> just add the commands that redirect to a file, then vcomplete those files in post_install

I assumed you mean that and I tried, but the options for khal that are autocompleted doesn't
even work... I've also installed directly into a virtualenv with pip to check if the behavior
is different. But it's the same. The completed options are not implemented in khal.

dataCobra commented:
This software does need further testing. I don't know yet what is causing the problems.

I'll investigate further.

Anachron commented:
This looks fine to me, do you want me to test it or just wanted to notify me?

classabbyamp commented:
this seems to give completions files that work (tested zsh)

```diff
diff --git a/srcpkgs/khal/template b/srcpkgs/khal/template
index 6b74897878..2f6331ed31 100644
--- a/srcpkgs/khal/template
+++ b/srcpkgs/khal/template
@@ -3,10 +3,10 @@ pkgname=khal
 version=0.10.5
 revision=1
 build_style=python3-module
-hostmakedepends="python3-setuptools"
-depends="python3-setuptools python3-click python3-click-log python3-configobj
+hostmakedepends="python3-setuptools python3-click python3-click-log python3-configobj
  python3-dateutil python3-icalendar python3-pytz python3-tzlocal
  python3-urwid python3-xdg python3-atomicwrites"
+depends="$hostmakedepends"
 checkdepends="python3-pytest python3-freezegun vdirsyncer $depends"
 short_desc="Command-line calendar build around CalDAV"
 maintainer="Anachron <gith@cron.world>"
@@ -25,8 +25,9 @@ pre_build() {
 
 post_install() {
        vlicense COPYING
-       vcompletion "${FILESDIR}/khal.bash" bash
-       vcompletion "${FILESDIR}/khal.fish" fish
-       vcompletion "${FILESDIR}/khal.zsh" zsh
+       for sh in bash fish zsh; do
+               env PYTHONPATH=$DESTDIR/$py3_sitelib _KHAL_COMPLETE="${sh}_source" $DESTDIR/usr/bin/khal > "khal.${sh}"
+               vcompletion "khal.${sh}" $sh
+       done
        vsconf khal.conf.sample
 }
 ```

dataCobra commented:
Well, I should've used a testuser for this purpose and not my main user. The dotfiles on my
main user where the reason for the problems while testing...

Lesson learned. I'll use a testuser from now on.

dataCobra commented:
Thanks a lot for the help. I've learned so much while updating this package.

I pushed the final version of template now.

dataCobra commented:
The next step is to get the creation of the manpage working for khal.

At the moment I've no idea how to accomplish that.

For that I'll create a new PR.

classabbyamp commented:
```
cd doc
make man
```
should do it, but you may need some more hostmakedepends for it to work

dataCobra commented:
I've tried to add this in, but Sphinx can't find the khal module when it tries to import.
```diff
diff --git a/srcpkgs/khal/template b/srcpkgs/khal/template
index 2f6331ed31..f06be57363 100644
--- a/srcpkgs/khal/template
+++ b/srcpkgs/khal/template
@@ -3,10 +3,13 @@ pkgname=khal
 version=0.10.5
 revision=1
 build_style=python3-module
-hostmakedepends="python3-setuptools python3-click python3-click-log python3-configobj
- python3-dateutil python3-icalendar python3-pytz python3-tzlocal
- python3-urwid python3-xdg python3-atomicwrites"
-depends="$hostmakedepends"
+hostmakedepends="python3-setuptools python3-Sphinx python3-sphinxcontrib
+ python3-click python3-click-log python3-configobj python3-dateutil
+ python3-icalendar python3-pytz python3-tzlocal python3-urwid python3-xdg
+ python3-atomicwrites"
+depends="python3-click python3-click-log python3-configobj python3-dateutil
+ python3-icalendar python3-pytz python3-tzlocal python3-urwid python3-xdg
+ python3-atomicwrites"
 checkdepends="python3-pytest python3-freezegun vdirsyncer $depends"
 short_desc="Command-line calendar build around CalDAV"
 maintainer="Anachron <gith@cron.world>"
@@ -30,4 +33,6 @@ post_install() {
 		vcompletion "khal.${sh}" $sh
 	done
 	vsconf khal.conf.sample
+        cd doc/
+        make man
 }
```

The following error message is stopping the process:
```python
sphinx-build -b man -d build/doctrees   source build/man
Running Sphinx v5.0.2

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/sphinx/config.py", line 343, in eval_config_file
    exec(code, namespace)
  File "/builddir/khal-0.10.5/doc/source/conf.py", line 17, in <module>
    import khal
ModuleNotFoundError: No module named 'khal'

make: *** [Makefile:131: man] Error 2
=> ERROR: khal-0.10.5_1: post_install: 'make man' exited with 2
=> ERROR:   in post_install() at srcpkgs/khal/template:37
```

dataCobra commented:
I got it working!

But it has a dependency which is not yet available in remote repositories.

Should I add the package to this PR? PKG: python3-sphinxcontrib-newsfeed

Anachron commented:
Did you add `PYTHONPATH` as in `Arch Linux`s https://github.com/archlinux/svntogit-community/blob/packages/khal/trunk/PKGBUILD#L31?

dataCobra commented:
I've created a separate PR #38602 for the new host make dependency now.

I've successful tested the building process with khal and python3-sphinxcontrib-newsfeed.

I also use the packages now already on my host machine.

paper42 commented:
add that commit here, it doesn't make sense separately from this PR and this PR doesn't work without it

dataCobra commented:
> add that commit here, it doesn't make sense separately from this PR and this PR doesn't work without it

I've done it as you said. :)

dataCobra commented:
Hey @Anachron,

does this still look fine to you? :slightly_smiling_face:

Anachron commented:
Yes @dataCobra , excellent work!

dataCobra commented:
Ups, I did git magic and messed up my branch...

I'll fix it and send it again.

dataCobra commented:
Alright, everything is again as it should be. :+1:

dataCobra commented:
If something is missing to get it merged, just let me know. :slightly_smiling_face:

classabbyamp commented on srcpkgs/python3-sphinxcontrib-newsfeed/template:
```diff
@@ -0,0 +1,20 @@
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3"
+checkdepends="python3-Sphinx"
```
there are no tests, so this and `make_check=ci-skip` should be removed

classabbyamp commented on srcpkgs/python3-sphinxcontrib-newsfeed/template:
```diff
@@ -0,0 +1,20 @@
+short_desc="Adding a Blog, News or Announcements section to a Sphinx website"
+maintainer="Orphaned <orphan@voidlinux.org>"
+license="BSD-2-Clause"
+homepage="http://sphinx-doc.org"
```
homepage is https://github.com/prometheusresearch/sphinxcontrib-newsfeed

classabbyamp commented on srcpkgs/python3-sphinxcontrib-newsfeed/template:
```diff
@@ -0,0 +1,20 @@
+wrksrc=sphinxcontrib-newsfeed-${version}
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3"
```
```suggestion
depends="python3-Sphinx"
```
https://github.com/prometheusresearch/sphinxcontrib-newsfeed/blob/0.1.4/setup.py#L28

classabbyamp commented:
if there is, it does not have any indication of it in the repo

paper42 commented:
It only checks if all required packages are installed and fails because Sphinx is not installed, add it to $checkdepends with `checkdepends="$depends" `.

dataCobra commented:
Hey @paper42,

> It only checks if all required packages are installed and fails because Sphinx is not installed, add it to $checkdepends with `checkdepends="$depends" `.

I've added Sphinx, but it still fails because of a TypeError. I've tried to get rid of it
but was not able to find a solution.

classabbyamp commented:
even when adding the `test_suite` argument to setup.py, `setup.py test` fails.

fwiw, sphinx doesn't even *need* `sphinxcontrib-newsfeed` for generating the manpage. with this patch and removal of `sphinx_rtd_theme` (which is just for the html docs) from the hostmakedepends it still generates the same manpages. There are warnings but they don't affect the output. 

```patch
--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -107,7 +107,6 @@
     'sphinx.ext.autodoc',
     'sphinx.ext.intersphinx',
     'sphinx.ext.todo',
-    'sphinxcontrib.newsfeed',
 ]
 
 # Add any paths that contain templates here, relative to this directory.
 ```

dataCobra commented:
I've added the patch and and changed the khal template accordingly. I've also removed the `sphinx_rtd_theme` dependency. After that I successful build the package.

The push does no longer contain the `python3-sphinxcontrib-newsfeed` package.
